server/auth/routes.js

```diff
@@ -7,48 +7,136 @@ const db = require("../services/database/users");
  * Users Login
  */
 router.post("/login", async (req, res, next) => {
-	const { email, password } = req.body;
-	console.log(`Login attempt ${email}`);
-	try {
-		const user = await db.getUserByEmail(email);
-
-		if (!email || !password || !user || password !== user.password) {
-			return res.sendStatus(403);
-		}
-
-		const secret = process.env.JWT_SECRET || "your_jwt_secret";
-		const token = jwt.sign({ userId: user.id }, secret);
-		delete user.password;
-		delete user.salt;
-		return res.send({ token, user });
-	} catch (e) {
-		console.error(e);
-		next(e);
-	}
+  const { email, password } = req.body;
+  console.log(`Login attempt ${email}`);
+  try {
+    const user = await db.getUserByEmail(email);
+
+    if (!email || !password || !user || password !== user.password) {
+      return res.sendStatus(403);
+    }
+
+    const secret = process.env.JWT_SECRET || "your_jwt_secret";
+    const token = jwt.sign({ userId: user.id }, secret);
+    delete user.password;
+    delete user.salt;
+    return res.send({ token, user });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
 });
 
 /**
  * Users Registration
  */
 router.post("/register", async (req, res, next) => {
-	const { email, password } = req.body;
-
-	const user = {
-		email,
-		password
-	};
-
-	db.createUser(user)
-		.then(() => {
-			res.send({
-				success: true,
-				message: "Account created",
-			});
-		})
-		.catch((err) => {
-			console.log(err);
-			next(err);
-		});
+  const { email, password, name, role } = req.body;
+
+  const user = {
+    email,
+    password,
+    name,
+    role
+  };
+
+  //   if (
+  //     user.email != null &&
+  //     user.email.length > 1 &&
+  //     user.password != null &&
+  //     user.password.length > 1 &&
+  //     user.name != null &&
+  //     user.name.length > 1 &&
+  //     user.role != null &&
+  //     user.role.length > 1 &&
+  //     (user.role == "Interpreter" || user.role == "User")
+  //   ) {
+
+  //   } else {
+  //     res.status(400).send({
+  //       success: false,
+  //       message: "Empty Validation"
+  //     });
+  //   }
+
+  if (
+    validateEmailAddress(email) &&
+    validateName(name) &&
+    validatePassword(password) &&
+    validateRole(role)
+  ) {
+    db.createUser(user)
+      .then(() => {
+        res.send({
+          success: true,
+          message: "Account created"
+        });
+      })
+      .catch(err => {
+        res.status(400).send({
+          success: false,
+          message: err
+        });
+        console.log(err);
+        next(err);
+      });
+  } else {
+    res.send(err);
+  }
+  function validateEmailAddress(email) {
```

Comment:
Can you define all these validator functions in a different files rather than inside the route implementation? You can create a file helpers.js or validators.js and put all these functions inside.

Reply:
I created new file and moved all the validation functions there